src/App.tsx

```diff
@@ -1,26 +1,124 @@
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [hasError, setHasError] = useState<Errors>(Errors.NoError);
+  const [filterBy, setFilterBy] = useState<FilterTodosBy>(FilterTodosBy.All);
+  const [title, setTitle] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isDeleting, setIsDeleting] = useState<boolean>(false);
+  const [isDeleteAllCompleted, setIsDeleteAllCompleted] =
+    useState<boolean>(false);
+
+  useEffect(() => {
+    setHasError(Errors.NoError);
+    let timer: NodeJS.Timeout | undefined;
+
+    (async () => {
+      try {
+        const response = await getTodos();
+
+        setTodos(response);
+      } catch {
+        setHasError(Errors.UnableToLoad);
+        timer = setTimeout(() => {
+          setHasError(Errors.NoError);
+        }, 3000);
+      }
+    })();
+
+    return () => {
+      clearTimeout(timer);
+    };
+  }, []);
+
+  const filteredTodos = todos.filter(todo => {
+    switch (filterBy) {
+      case FilterTodosBy.Active:
+        return !todo.completed;
+      case FilterTodosBy.Completed:
+        return todo.completed;
+      default:
+        return true;
+    }
+  });
+
+  const uncompletedTodos = todos.filter(todo => !todo.completed).length;
+  const completedTodosLenght = todos.filter(todo => todo.completed).length;
+
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  const handleDeleteAllCompleted = async () => {
+    setIsDeleteAllCompleted(true);
+    const completedTodos = todos.filter(todo => todo.completed);
+
+    try {
+      for (const todo of completedTodos) {
+        try {
+          await deleteTodo(todo.id);
+          setTodos(currTodos =>
+            currTodos.filter(currTodo => currTodo.id !== todo.id),
+          );
+        } catch (error) {
+          setHasError(Errors.UnableToDelete);
+        }
+      }
+    } finally {
+      setIsDeleteAllCompleted(false);
+    }
+  };
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">
-          React Todo App - Load Todos
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <div className="todoapp__content">
+        <h1 className="todoapp__title">todos</h1>
+        <Header
+          title={title}
+          setTitle={setTitle}
+          setHasError={setHasError}
+          hasError={hasError}
+          USER_ID={USER_ID}
+          todos={todos}
+          setTodos={setTodos}
+          setTempTodo={setTempTodo}
+          tempTodo={tempTodo}
+          inputRef={inputRef}
+          isDeleteAllCompleted={isDeleteAllCompleted}
+        />
+        <TodoList
```

Comment:
too many props. Move handleDelete to the parent component

Reply:
I've made all the specified changes and transferred them to part 3.